lib/std/mem.zig

```diff
@@ -1459,8 +1463,13 @@ pub fn writeIntSliceLittle(comptime T: type, buffer: []u8, value: T) void {
 pub fn writeIntSliceBig(comptime T: type, buffer: []u8, value: T) void {
     assert(buffer.len >= @divExact(@typeInfo(T).Int.bits, 8));
 
-    if (@typeInfo(T).Int.bits == 0)
+    if (@typeInfo(T).Int.bits == 0) {
         return set(u8, buffer, 0);
+    } else if (@typeInfo(T).Int.bits == 8) {
+        set(u8, buffer, 0);
```

Comment:
Do you need to `set` with `255` if `value` is negative?

Reply:
I hadn't even thought of that, I was just basing my results off the negative i16 test where the rest of the values are set to zero, but maybe they should be set to 0xff. I guess it really depends on what the intention of #8865 was when it was implemented